### ISSUE TITLE
PeriodicTask on iOS

### ISSUE THREAD

**Issue by LostInDarkMath** [fluttercommunity/flutter_workmanager#306]
Hi there,
I now that there are lots of other issues where this question is discussed. I just want to ask to ask what is the best way to create a periodic task on iOS?

## What I know
- The current documentation says, that this is not supported.
- https://github.com/fluttercommunity/flutter_workmanager/issues/295 says, that is could be supported soon 
- https://github.com/fluttercommunity/flutter_workmanager/issues/109 also raises hope
- In https://github.com/fluttercommunity/flutter_workmanager/issues/206 is said, that small batches are no problem. But I do not found an example. I just want to make an API-Request every 15 minutes like on Android. How can I achieve this?

So I'm very confused at the moment. 
1. Does the library support periodic background tasks on iOS **now**? How?
2. If not: Is this feature in development or at least on the road map?

Thank you very much in advance!

**Comment by karthisu** [comment]
Same situation. iOS - PlatformException(unhandledMethod("registerPeriodicTask")

**Comment by Stratovarius93** [comment]
Same situation, has anyone been able to solve it ?
I'm using this alternative to run in Android but not iOS.
```
Future<void> _fetchBackgrountLocation(
    Function callbackDispatcher, String fetchBackground) async {
  await Workmanager().initialize(
    callbackDispatcher,
    isInDebugMode: true,
  );

  if (Platform.isAndroid) {
    Workmanager().registerPeriodicTask(
      "1",
      fetchBackground,
      frequency: Duration(minutes: 15),
    );
  }
}
```

**Comment by arbazKnarkz** [comment]
Same here, any update on this?

**Comment by arbazKnarkz** [comment]
Most of the things says [Android Only] so i think this plugin is only for Android and not for the IOS

**Comment by ened** [comment]
Part of the issue is the plugins current APIs.

`registerPeriodicTask` is indeed not working on iOS, because there is no equivalent.

The new BG task APIs support scheduling background work and once they are executed, you can schedule another piece of work with a minimumDelay, thus causing the `illusion` of periodic work. A lot of this is due to how iOS schedules background work.

PRs on documenting, testing etc are very welcome. The code is all visible in `0.5.0-dev.8`.

**Comment by LostInDarkMath** [comment]
Thanks for your answer!

Could you please provide a minimal example of an iOS background task that schedules another which schedules another one and so on?

And I personally think that it would be a good idea to use *illusional periodic task* for the `registerPeriodicTask` function on iOS. Even if this implementation might not be perfect it is still better than no implementation at all imho.

**Comment by cormalenv** [comment]
> Part of the issue is the plugins current APIs.
> 
> `registerPeriodicTask` is indeed not working on iOS, because there is no equivalent.
> 
> The new BG task APIs support scheduling background work and once they are executed, you can schedule another piece of work with a minimumDelay, thus causing the `illusion` of periodic work. A lot of this is due to how iOS schedules background work.

For others searching for an understanding of the various iOS tasking options, intentions and limitations there is an excellent video by Apple explaining as much [here](https://developer.apple.com/videos/play/wwdc2020/10063/).

**Comment by saad197** [comment]
Man forget periodic, I cannot even do OneOff task on iOS.

**Comment by cormalenv** [comment]
> Man forget periodic, I cannot even do OneOff task on iOS. Hope they fix this soon.

https://github.com/fluttercommunity/flutter_workmanager/issues/357#issuecomment-1028842173

**Comment by lyio** [comment]
@LostInDarkMath, I have to agree that the documentation is confusing. I am also not quite sure, which things are supported on both platforms.

Maybe someone who knows the intricacies of the plugin implementation could go over the documentation (and the comments in the code) and see that it accurately reflects what the plugin is doing.

**Comment by bdshadow** [comment]
it looks like `flutter_local_notifications` plugin has the ability to create periodic notifications in iOS: https://pub.dev/packages/flutter_local_notifications#periodically-show-a-notification-with-a-specified-interval. Also a short example of it [here](https://stackoverflow.com/questions/61905143/how-to-create-alarm-app-in-flutter-for-ios).

Why can't it be done the same here in the workmanager?
(sorry, I'm not an iOS developer and may be missing smth)

**Comment by xunreal75** [comment]
Scheduled notifications and bgAppRefreshTask are very different.

You can schedule a notification.
This notification will displayed at scheduled time.

A background work is called by iOS itself and has no possibility to schedule a specific time. The call  is randomized by iOS- see apple documents for bgAppRefreshTask

**Comment by Myzel394** [comment]
@Stratovarius93 did you find a solution for iOS?

**Comment by ziqq** [comment]
Hello there, any news?

**Comment by junaidirshad098** [comment]
[VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(unhandledMethod("registerPeriodicTask") error, Unhandled method registerPeriodicTask, null, null)

I'm facing this issue in IOS, android is working fine.

**Comment by mariomurrent-softwaresolutions** [comment]
Any updates on this?

**Comment by absar** [comment]
Check https://github.com/fluttercommunity/flutter_workmanager/pull/511 which has major enhancements for iOS

**Comment by ened** [comment]
Quick status update (2026-08): iOS periodic execution is now documented honestly — see the platform capability matrix in the docs (PR #688) and the Quick Start.

The short version:
- On iOS, "periodic" means *best-effort background fetch*: `registerPeriodicTask` submits a `BGAppRefreshTaskRequest` with your identifier (which must be in Info.plist and registered in AppDelegate). iOS then decides when (and whether) it runs, based on usage patterns — there is no guaranteed interval.
- For exact timing, schedule one-off/processing tasks and chain the next one from inside the callback.
- `unhandledMethod("registerPeriodicTask")` in the thread above is from very old plugin versions (method-channel rewrite landed in 0.9.0) — upgrade and clean-rebuild.

Android remains fully reliable for periodic work (15-min minimum).

**Comment by ened** [comment]
iOS periodic behavior is now documented honestly in the platform capability matrix (`docs/index.mdx`, added in #688) and the Quick Start; the new iOS background-strategy comparison (BGAppRefresh vs BGProcessing vs BGHealthResearch vs BGContinuedProcessing) also clarifies the 15-minute best-effort model. Short answer: exact-interval background work is not possible on iOS — see the matrix. Closing as answered.

**Comment by chris-wickens** [comment]
Thanks for clarifying.

> For exact timing, schedule one-off/processing tasks and chain the next one from inside the callback.

Are you saying this is a reliable alternative? Are there any downsides to doing this instead of periodic callbacks?
